sklearn/metrics/tests/test_ranking.py

```diff
@@ -1468,3 +1469,62 @@ def test_partial_roc_auc_score():
         assert_almost_equal(
             roc_auc_score(y_true, y_pred, max_fpr=max_fpr),
             _partial_roc_auc_score(y_true, y_pred, max_fpr))
+
+
+@pytest.mark.parametrize('y_true, k, normalize, sample_weight, true_score', [
+    ([0, 1, 2, 3, 4], 2, True, None, .2),
+    ([0, 1, 2, 3, 4], 3, True, None, .4),
+    ([0, 1, 2, 3, 4], 4, True, None, .8),
+    ([0, 1, 2, 3, 4], 2, False, None, 1),
+    ([0, 1, 2, 3, 4], 3, False, None, 2),
+    ([0, 1, 2, 3, 4], 4, False, None, 4),
+    ([0, 0, 0, 0, 0], 2, True, None, 1),
+    ([0, 1, 2, 3, 4], 2, True, [4, 1, 1, 1, 1], .5),
+    ([0, 1, 2, 3, 4], 3, False, [1, 2, 1, 1, 1], 3),
+])
+def test_top_k_accuracy_score(y_true, k, normalize, sample_weight, true_score):
+    y_score = np.array([
+        [0.4, 0.3, 0.1, 0.1, 0.1],
+        [0.3, 0.2, 0.3, 0.1, 0.1],
+        [0.3, 0.4, 0.1, 0.2, 0.],
+        [0.3, 0.4, 0.2, 0.1, 0.],
+        [0.4, 0.1, 0.2, 0.25, 0.05],
+    ])
+    assert top_k_accuracy_score(y_true, y_score, k, normalize,
+                                sample_weight) == pytest.approx(true_score)
+
+
+@pytest.mark.parametrize('y_true, k, msg', [
+    (
+        [0, 1, 2],
+        1,
+        "'k'=1 is equivalent to 'metrics.accuracy_score'. Please, use that "
+        "method instead."
+    ),
+    (
+        [.57, 1, 2],
+        2,
+        "'y_true' type must be 'binary' or 'multiclass', got 'continuous' "
+        "instead."
+    ),
+    (
+        [0, 1, 2],
+        3,
+        r"'k' \(3\) can't be greater than or equal to 'n_classes' \(3\). Will "
+        "result in a perfect score and is therefore meaningless."
```

Comment:
Just FYI, you don't need to catch the whole message. The relevant part is enough

```suggestion
        r"'k' \(3\) can't be greater than or equal to 'n_classes' \(3\). Will "
        "result in a perfect score and is therefore meaningless."
```
```suggestion
        r"'k' \(3\) can't be greater than or equal to 'n_classes' \(3\)"
```

Reply:
Nice! Didn't know it was possible, thx!